python/ctsm/subset_data.py

```diff
@@ -403,6 +413,43 @@ def check_args(args):
         )
         raise argparse.ArgumentError(None, err_msg)
 
+    if args.create_landuse and not args.create_surfdata:
+        err_msg = textwrap.dedent(
+            """\
+                \n ------------------------------------
+                \n --create-landuse option requires the --create-surface option:
+                """
+        )
+        raise argparse.ArgumentError(None, err_msg)
+    if args.surf_year != 2000 and not args.create_surfdata:
+        err_msg = textwrap.dedent(
+            """\
+                \n ------------------------------------
+                \n --surf-year option is set to something besides the default of 2000
+                \n without the --create-surface option"
+                """
+        )
+        raise argparse.ArgumentError(None, err_msg)
+
+    if args.surf_year != 1850 and args.create_landuse:
+        err_msg = textwrap.dedent(
+            """\
+                \n ------------------------------------
+                \n --surf-year option is NOT set to 1850 and the --create-landuse option
+                \n is selected which requires it to be 1850
+                """
```

Comment:
- [ ] Why is this required to be 1850? Some additional text in the error message would be helpful, explaining what happens when you do set it to 1850 (you still get a landuse timeseries file with 1850-2015).
- [ ] When I did set this to 1850, I still got an fsurdat file for 2000.

Reply:
On the first. When you have a landuse timeseries file for a year range of XXXX-YYYY, the surface dataset HAS to be the beginning of the range XXXX. This enables the model to do internal consistency checking to make sure the two files go together.

Reply:
On the second, there is a config file that stores the location for the files that is separate from build-namelist XML and it's currently setup to be hardcoded to either 1850 or 2000. We do have a different file on ctsm5.2 for 1850, with the default being 2000. It would be good to handle this differently. But we ought to brainstorm on what it should look like, as well as needing to spend time to work on it.

Reply:
> On the first. When you have a landuse timeseries file for a year range of XXXX-YYYY, the surface dataset HAS to be the beginning of the range XXXX. This enables the model to do internal consistency checking to make sure the two files go together.

I guess my thought is that `--surf-year` should be allowed to be set to 2000 with `--create-landuse`, but that the resulting landuse timeseries file in that case would be 2000-2015.

> On the second, there is a config file that stores the location for the files that is separate from build-namelist XML and it's currently setup to be hardcoded to either 1850 or 2000. We do have a different file on ctsm5.2 for 1850, with the default being 2000. It would be good to handle this differently. But we ought to brainstorm on what it should look like, as well as needing to spend time to work on it.

That doesn't really explain why, when I specified `--surf-year=1850` in the call of `subset_data`, I got an fsurdat file for 2000. Not only is `simyr2000` in the filename, the files' data are identical.

Reply:
Hmmm. We might need to go over some of this together to see exactly what you are seeing.

OK, interesting idea on 2000-2015, although I don't think it truncates the years for the landuse timeseries file. But, I could be wrong. And it could be added also. That actually might be a useful feature for single-point cases because you could have it be over just the period that you have your tower site data for.

This is definitely a pain point that could definitely use some work to get it to work more intuitively...

Reply:
Right, it probably doesn't truncate the years, but that behavior could be added.

For the result I'm describing, I did the following:
```bash
conda run -n ctsm_pylib \
tools/site_and_regional/subset_data region \
    --create-landuse --create-surface \
    --surf-year=1850 \
    --create-domain \
    --lat1 -15 --lat2 -11 \
    --lon1 -60 --lon2 -52 \
    --outdir /glade/work/samrabin/arc_of_deforestation/mato_grosso \
    --verbose --overwrite
```
and got the `*_c240313.nc* files in `/glade/work/samrabin/arc_of_deforestation/mato_grosso/`.